### PR TITLE
Members-only Project

### PR DESCRIPTION
**Pull request by GraceOyiza**
- All required changes fixed

### CODE REVIEW

File: app/views/posts/index.html.erb [hunk 1]
```diff
@@ -0,0 +1,13 @@
...
+<%= link_to 'Sign Up', new_user_registration_path%>
+<%= link_to 'Create post',  new_post_path%>
+<h1> POST LIST </h1>
+
+<% @post.each do|post| %>
+  <h2> <%= post.title %></h2>
+  <p> <%= post.body %></p>
+ <% if current_user %> 
+  <%=  post.user.username %> 
+<% end %>
```
Comment (kobiyoyo) [inline app/views/posts/index.html.erb]: - [ ] Please remove all logic from your views. There should be absolutely no logic in views. Instead, use helper methods to outsource view logic.

File: app/controllers/posts_controller.rb [hunk 2]
```diff
@@ -0,0 +1,29 @@
+class PostsController < ApplicationController
+  before_action :authenticate_user!, only: %i[new create]
+
+  def index
+    @post = Post.all
+  end
+
+  def new
+    @post = Post.new
+    puts current_user.id
```
Comment (kobiyoyo) [inline app/controllers/posts_controller.rb]: Please remove this, as it is not required in this application.

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [ ] Remove the `README` title in readme.md file
- [ ] Pull Request should have a meaningful description
- [ ] Lets remove commented codes
- [ ] Logout should display only when user is logged in
# \[OPTIONAL\] 

* [ ] A helper method to check if user is signed_in? can be handy here

```ruby
# must be a helper method to be used in views
helper_method :signed_in?
def signed_in?
  !current_user.nil?
end
```
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

**Comment by GraceOyiza** [comment]
- This project is about members authentication, Post and User models. 
- The association between the post and user were defined.

File: app/helpers/posts_helper.rb [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module PostsHelper
+  def params_helper
+    params.require(:post).permit(:title, :body)
+  end
```
Comment (kobiyoyo) [inline app/helpers/posts_helper.rb]: - [ ] Kindly remove this as you already have it in controllers

**Review by kobiyoyo — CHANGES_REQUESTED** [review]
## Hi Team    
#### Great Job so far, everything works fine, but there are a few more things required. 
### General 
- [ ] Logout should display only when a user is logged in (its the same implementation for author)
- check comments below
### Make changes and submit another code review request.
*Happy coding* 
(Any questions Feel free to contact me on slack @kobiyoyo)

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_many :posts
+
+  validates :username, presence: true, uniqueness: true, length: { minimum: 5, maximum: 30 }
+  validates :email, presence: true, case_sensitive: false, length: { minimum: 5, maximum: 30 }
+  validates :password, presence: true, length: { minimum: 5, maximum: 30 }
+  # Include default devise modules. Others available are:
+  # :confirmable, :lockable, :timeoutable, :trackable and :omniauthable
```
Comment (kobiyoyo) [inline app/models/user.rb]: - [ ] Lets delete this commented code

**Comment by GraceOyiza** [comment]
- All required errors fixed

**Review by tirthajyoti-ghosh — APPROVED** [review]
# Approved ✅

## You have done an incredible job making this project 👏🧨💪! Just perfect 💯! 

### You have met all the requirements of the project. Go ahead and merge it! Onto the next project 🚀 !

Happy coding 👨‍💻 

<hr>

**Tirthajyoti Ghosh**

[GitHub](https://github.com/tirthajyoti-ghosh) | [LinkedIn](https://www.linkedin.com/in/tirthajyoti-ghosh/) | [Twitter](https://twitter.com/terrific_ghosh)

**Comment by Alaska01** [comment]
Project has been approved!!!!

**Pull request merged**